recipe/meta.yaml

```diff
@@ -19,22 +19,22 @@ requirements:
     - python                                 # [build_platform != target_platform]
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - cython                                 # [build_platform != target_platform]
-    - numpy                                  # [build_platform != target_platform]
+    - numpy >=1.24                           # [build_platform != target_platform]
     - {{ compiler('c') }}
     - {{ stdlib("c") }}
     - {{ compiler('cxx') }}
   host:
     - python
     - pip
     - cython
-    - numpy
-    - libgdal
+    - numpy >=1.24
```

Comment:
> I am unsure about pinning numpy within conda-forge as conda-forge seems to handle that...

Had a look at https://anaconda.org/conda-forge/rasterio/files?version=1.3.11, and it seems to be pinned to `numpy >=1.19,<3`. I'm a little confused because looking at https://github.com/h-vetinari/conda-forge-pinning-feedstock/blame/5daae1321d6e714ac872b12d06f1882a6bbb034e/recipe/conda_build_config.yaml#L554-L559, the pin should be on `numpy >=1.17` instead? There's some discussion of this too at https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/4816.

I think we should stick with what `rasterio=1.4.0` is doing upstream and set the minimum pin explicitly here.